libraries/common/list.effekt

```diff
@@ -261,6 +339,27 @@ def foldRight[A, B](l: List[A], init: B) { f: (A, B) => B }: B = {
   acc
 }
 
+/// Fold and map a list with a delimiter, a prefix, a suffix,
+/// concatenating the result given the `combine` function.
+///
+/// Example:
+/// ```
+/// > [1, 2, 3].foldMapDelimited("[", ", ", "]") { n => show(n) } { (s1, s2) => s1 ++ s2 }
+/// "[1, 2, 3]"
+/// ```
+///
+/// O(N)
+def foldMapDelimited[A, B](list: List[A], prefix: B, delimiter: B, suffix: B) { convert: A => B } { combine: (B, B) => B }: B = {
+  def go(acc: B, as: List[A]): B = as match {
+    case Nil() => acc
+    case Cons(head, Nil()) => combine(acc, convert(head))
+    case Cons(head, tail) =>
+      val res = combine(combine(acc, convert(head)), delimiter)
+      go(res, tail)
+  }
+  combine(go(prefix, list), suffix)
+}
```

Comment:
I can't use `acc.combine(...)` here (cc @b-studios)
I thought I already reported this, but I can't find the issue 😓.

<img width="605" alt="Screenshot 2025-02-21 at 20 37 41" src="https://github.com/user-attachments/assets/8160b47b-e683-480f-88d1-70d862561b2f" />

Reply:
By the way, feel free to bikeshed the name of this function -- I don't really know what to call it, but it's pretty useful for simple custom list pretty-printing.